From db9e0de64102e2fa0b60774135da64085ec9286a Mon Sep 17 00:00:00 2001
From: "jianxin.xu" <jianxin.xu@sophgo.com>
Date: Wed, 18 Jun 2025 18:48:51 +0800
Subject: [PATCH 1/2] [fix](reboot):fix reboot in cv_pm.c [description ]:NA
 [root   cause]:NA [JIRA      ID]:NA [chip project]:NA [side effects]:NA
 [Affected *.ko/*.so/*.a] :NA

Change-Id: I2ba58901970af5cd96bc8cf51f4488ac5a598344
---
 plat/cvitek/cv184x/common/cv_pm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plat/cvitek/cv184x/common/cv_pm.c b/plat/cvitek/cv184x/common/cv_pm.c
index 8c53bd7..3d00388 100644
--- a/plat/cvitek/cv184x/common/cv_pm.c
+++ b/plat/cvitek/cv184x/common/cv_pm.c
@@ -456,11 +456,11 @@ static void __dead2 bm_system_reset(void)
 	mmio_setbits_32(TPUSYS_ENABLE, 0x1);
 	memset((void *)BM_IO_BUF_BASE, 0x0, BM_IO_BUF_SIZE); //8KB
 
-	mmio_write_32(REG_RTC_BASE + RTC_EN_PWR_CYC_REQ, 0x01);
-	while (mmio_read_32(REG_RTC_BASE + RTC_EN_PWR_CYC_REQ) != 0x01)
+	mmio_write_32(REG_RTC_BASE + RTC_EN_WARM_RST_REQ, 0x01);
+	while (mmio_read_32(REG_RTC_BASE + RTC_EN_WARM_RST_REQ) != 0x01)
 		;
 	mmio_write_32(REG_RTC_CTRL_BASE + RTC_CTRL0_UNLOCKKEY, 0xAB18);
-	mmio_setbits_32(REG_RTC_CTRL_BASE + RTC_CTRL0, 0xFFFF0800 | (0x1 << 3));
+	mmio_setbits_32(REG_RTC_CTRL_BASE + RTC_CTRL0, 0xFFFF0800 | (0x1 << 4));
 	panic();
 }
 
-- 
2.25.1

